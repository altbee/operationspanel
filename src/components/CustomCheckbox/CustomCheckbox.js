import React, { Component } from "react";
import PropTypes from "prop-types";

class CustomCheckbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isCheckedFromProps: !!props.isChecked
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(prevState => ({
      isCheckedFromProps: !prevState.isCheckedFromProps
    }));
  }

  render() {
    const { isChecked, number, label, inline, ...rest } = this.props;
    const { isCheckedFromProps } = this.state;
    const classes = inline ? "checkbox checkbox-inline" : "checkbox";
    return (
      <div className={classes}>
        <label htmlFor={number}>
          <input
            id={number}
            type="checkbox"
            onChange={this.handleClick}
            checked={isCheckedFromProps}
            {...rest}
          />
          {label}
        </label>
      </div>
    );
  }
}

CustomCheckbox.propTypes = {
  isChecked: PropTypes.bool,
  number: PropTypes.number,
  label: PropTypes.string,
  inline: PropTypes.bool
};

CustomCheckbox.defaultProps = {
  isChecked: false,
  number: 0,
  label: "",
  inline: false
};

export default CustomCheckbox;
