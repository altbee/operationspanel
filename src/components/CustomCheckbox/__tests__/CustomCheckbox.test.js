import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CustomCheckbox from "../CustomCheckbox";

configure({ adapter: new Adapter() });

describe("<CustomCheckbox /> ", () => {
  const definedProps = {
    isChecked: true,
    number: 150,
    label: "label text",
    inline: true
  };

  const rest = {
    disabled: true
  };

  it("renders properly", () => {
    const rendered = shallow(<CustomCheckbox />);
    expect(rendered.find("div").exists()).toBe(true);
    expect(rendered.find("div").hasClass("checkbox")).toBe(true);
    expect(rendered.find("div").hasClass("checkbox-inline")).toBe(false);
    expect(rendered.find("div").children().length).toEqual(1);
    expect(rendered.find("label").exists()).toBe(true);
    expect(
      rendered
        .find("div")
        .children()
        .children().length
    ).toEqual(1);
    expect(rendered.find("input").exists()).toBe(true);
  });

  it("accepts props and renders", () => {
    const rendered = shallow(<CustomCheckbox {...definedProps} />);
    expect(rendered.find("div").hasClass("checkbox-inline")).toBe(true);
    expect(
      rendered.find(`label[htmlFor=${definedProps.number}]`).exists()
    ).toBe(true);
    expect(rendered.find("label").text()).toEqual(definedProps.label);
    expect(rendered.find(`input[id=${definedProps.number}]`).exists()).toBe(
      true
    );
  });

  it("accepts {rest} object as props", () => {
    const rendered = shallow(<CustomCheckbox {...definedProps} {...rest} />);
    expect(rendered.find("[disabled=true]").exists()).toBe(true);
  });

  it("updates the state onChange", () => {
    const rendered = mount(<CustomCheckbox {...definedProps} />);
    expect(rendered.state().isCheckedFromProps).toBe(true);
    rendered.find('[type="checkbox"]').simulate("change");
    expect(rendered.state().isCheckedFromProps).toBe(false);
  });

  it("matches the snapshot", () => {
    const tree = shallow(<CustomCheckbox {...definedProps} />);
    expect(tree).toMatchSnapshot();
  });
});
