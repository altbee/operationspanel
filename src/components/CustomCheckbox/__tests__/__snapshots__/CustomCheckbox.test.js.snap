// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomCheckbox />  matches the snapshot 1`] = `
<div
  className="checkbox checkbox-inline"
>
  <label
    htmlFor={150}
  >
    <input
      checked={true}
      id={150}
      onChange={[Function]}
      type="checkbox"
    />
    label text
  </label>
</div>
`;
